src/rgw/rgw_admin.cc

```diff
@@ -360,6 +360,8 @@ void usage()
   cout << "   --secret/--secret-key=<key>       specify secret key\n";
   cout << "   --gen-access-key                  generate random access key (for S3)\n";
   cout << "   --gen-secret                      generate random secret key\n";
+  cout << "   --generate-key                    create user with or without credentials\n";
```

Comment:
maybe can we find a way to make `--gen-access-key=0 --gen-secret=0` do the right thing instead of adding a separate `--generate-key`?

Reply:
@cbodley the adminops have `--generate-key` flag and does not have `--gen-access-key` or `gen-secret` flags. @likid0 looking for similar changes like adminsop. To make it consistent IMO we need to drop one and choose other. @cbodley please share your inputs

Reply:
I think, as you mentioned, it would be great if we could give as much consistency as possible between the OPs and radosgw-admin CLI to avoid confusion among the users; the only thing about dropping one is taking care of the backward compatibility.

Reply:
it seems to me that making this consistent with the admin API is also making it inconsistent with itself. how does the new `--generate-key` interact with the existing `--gen-access-key` and `--gen-secret` options? can you please try to document this clearly?